ledger/notary-plugin/src/main/kotlin/net/corda/v5/ledger/notary/pluggable/NotarisationRequest.kt

```diff
@@ -0,0 +1,28 @@
+package net.corda.v5.ledger.notary.pluggable
+
+import net.corda.v5.crypto.SecureHash
+import net.corda.v5.ledger.utxo.StateRef
+
+/**
+ * A notarisation request specifies a list of states to consume and the id of the consuming transaction. Its primary
+ * purpose is for notarisation traceability – a signature over the notarisation request, [NotarisationRequestSignature],
+ * allows a notary to prove that a certain party requested the consumption of a particular state.
+ *
+ * While the signature must be retained, the notarisation request does not need to be transferred or stored anywhere - it
+ * can be built from a [net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction]. The notary can recompute it from
+ * the committed states index.
+ *
+ * Reference inputs states are not included as a separate property in the [NotarisationRequest] as they are not
+ * consumed.
+ *
+ * In case there is a need to prove that a party spent a particular state, the notary will:
+ * 1) Locate the consuming transaction id in the index, along with all other states consumed in the same transaction.
+ * 2) Build a [NotarisationRequest].
+ * 3) Locate the [NotarisationRequestSignature] for the transaction id. The signature will contain the signing public key.
+ * 4) Demonstrate the signature verifies against the serialized request. The provided states are always sorted internally,
+ *    to ensure the serialization does not get affected by the order.
+ */
```

Comment:
This is a really descriptive comment, but is going into a lot of (assumed) implementation detail - who's to say what an implementation will use, or what checks it will do?

These comments may still be valid in the context of an implementation class, but at the API level I would review this and make sure that this only refers to things that an API needs to worry about.

Reply:
Good point, I'll only keep the first paragraph, I think that's enough description for this interface.